### PR TITLE
Read docker endpoint from environment varibale DOCKER_HOST if present.

### LINKED ISSUE #14: Support alternative docker endpoints
Hi there - would you be open to a PR to allow ecs-init to read the location of the docker endpoint out of the config file?

### PR DESCRIPTION
**Pull request by tomwilkie**
Fixes #14

### CODE REVIEW

**Comment by tomwilkie** [comment]
Actually this doesn't work as DOCKER_HOST contains the unix:// prefix (typically).  I'll revise.

**Comment by samuelkarp** [comment]
Hi @tomwilkie,

Thanks for the pull request!  Apologies for my slow response; I was on vacation until today.

It looks like you're having some build failures with one of the tests: https://travis-ci.org/aws/amazon-ecs-init/builds/68507034#L121.  Note that you can run tests locally with `make test`.  Once you've addressed this and my line comments, we'd be happy to take your PR.

Please confirm that you can contribute this under the terms of the Apache License, Version 2.0.

Sam

File: ecs-init/docker/dependencies.go [hunk 1]
```diff
@@ -40,8 +46,15 @@ type _dockerclient struct {
 	docker *godocker.Client
 }
 
+func getDockerUnixSocket() string {
```
Comment (samuelkarp) [inline ecs-init/docker/dependencies.go]: Can you move this function (and associated constants) out to the `config` package?

File: ecs-init/docker/docker.go [hunk 2]
```diff
@@ -161,7 +161,7 @@ func (c *Client) loadEnvVariables() []string {
 
 func (c *Client) getHostConfig() *godocker.HostConfig {
 	binds := []string{
-		defaultDockerEndpoint + ":" + defaultDockerEndpoint,
+		dockerUnixSocket + ":" + dockerUnixSocket,
```
Comment (samuelkarp) [inline ecs-init/docker/docker.go]: Can you make sure that this gets mounted into the well-known location inside the container?

`config.DockerUnixSocket() + ":" + defaultDockerEndpoint`

**Comment by tomwilkie** [comment]
Thanks for the feedback @Samuel!; @2opremio is helping me out here, I got distracted with other things

**Comment by 2opremio** [comment]
@samuelkarp I made some changes to address your comments. Please review. I can squash them into a single commit once you are OK with them.

**Comment by samuelkarp** [comment]
Looks good to me, though I'd love to see a unit test for the new function.  Not required however, as I can add one afterward.

Can you confirm that you can contribute this under the terms of the Apache License, Version 2.0?

**Comment by 2opremio** [comment]
Yes, we can confirm that we can contribute this under the terms of the Apache License, Version 2.0. Thanks @samuelkarp

**Comment by samuelkarp** [comment]
Sorry for missing this earlier!  Can you recreate the pull request against the `dev` branch instead?

**Comment by 2opremio** [comment]
@samuelkarp sure! Closing in favor https://github.com/aws/amazon-ecs-init/pull/17

**Pull request closed without merging**